File: LeetCode/cpp/1725.cpp
//Link: https://leetcode.com/problems/number-of-rectangles-that-can-form-the-largest-square/

class Solution {
public:
    int countGoodRectangles(vector<vector<int>>& rectangles) {
        vector <int> sides;
        int side, maxLen = 0, count = 0;
        for(auto rect : rectangles) {
            side = (rect[0] < rect[1]) ? rect[0] : rect[1];
            sides.push_back(side);
            maxLen = (side > maxLen) ? side : maxLen;
        }
        for(int side : sides)
            if(side == maxLen) count++;
        return count;
    }
};

File: LeetCode/py/1725.py
#Link: https://leetcode.com/problems/number-of-rectangles-that-can-form-the-largest-square/

class Solution:
    def countGoodRectangles(self, rectangles: List[List[int]]) -> int:
        sides = []
        side = maxLen = 0
        for rect in rectangles:
            side = min(rect[0],rect[1])
            sides.append(side)
            maxLen = max(side, maxLen)
        return sides.count(maxLen)
        